plugins/BEdita/API/src/Controller/FoldersController.php

```diff
@@ -47,14 +48,15 @@ class FoldersController extends ObjectsController
      * `parent` relationship is valid and will return `parents` association.
      * `parents` relationship is not allowed.
      */
-    protected function findAssociation($relationship)
+    protected function findAssociation(string $relationship, ?Table $table = null): Association
     {
-        if ($relationship === 'parents') {
-            throw new NotFoundException(__d('bedita', 'Relationship "{0}" does not exist', $relationship));
-        }
-
-        if ($relationship === 'parent') {
-            return $this->Table->getAssociation('Parents');
+        if ($table === null) {
+            switch ($relationship) {
+                case 'parents':
+                    throw new NotFoundException(__d('bedita', 'Relationship "{0}" does not exist', $relationship));
+                case 'parent':
+                    return $this->Table->getAssociation('Parents');
+            }
         }
 
         return parent::findAssociation($relationship);
```

Comment:
shouldn't this pass `$table` along to the parent class?

Reply:
Ouch, I'm afraid you're right!
I'm fixing it and adding an integration test, if I can…